feat(tekton): support official bundle resolver (#19040)
renovatebot/renovate: lib/modules/manager/tekton/readme.md

The `tekton` manager focuses on providing updates to Tekton resource references.

Right now, Renovate's Tekton manager only supports references that are [Bundles](https://tekton.dev/docs/pipelines/tekton-bundle-contracts/).

> See the [docs](https://tekton.dev/docs/pipelines/resolution/) for the different kinds of Tekton references and their corresponding resolvers.

Reviewer: ```suggestion
Read the [Tekton Pipeline remote resolution docs](https://tekton.dev/docs/pipelines/resolution/) for the different kinds of Tekton references and their corresponding resolvers.
```

Let's use a clear link name.

### Using a Tekton Bundle reference

> There are three ways to use a Tekton Bundle reference:

Reviewer 2: is this still true? i only see two list items

Author: It should be:

1. Via the [Tekton Bundles Resolver](https://tekton.dev/docs/pipelines/bundle-resolver/)
1. Via the [`tektoncd/resolution` project](https://github.com/tektoncd/resolution)
1. Via the `taskRun.spec.taskRef.bundle` and the `pipelineRun.spec.pipelineRef.bundle` attributes

I believe that's what the diff says.

Reviewer 2: i don't know. i only see that it needs a third list items or revert to two. 🤷‍♂️😉

Author: Maybe I'm missing something... Here's what the rendered markdown file looks like:
![image](https://user-images.githubusercontent.com/5272931/204647601-979d8118-d21d-488e-be88-eeb1d767abe9.png)

> 1. Via the [Bundles Resolver](https://tekton.dev/docs/pipelines/bundle-resolver/)
> 2. Via the [`tektoncd/resolution` project](https://github.com/tektoncd/resolution)
> 3. Via the `taskRun.spec.taskRef.bundle` and the `pipelineRun.spec.pipelineRef.bundle` attributes

Reviewer: ```suggestion
1. Via the [Tekton Bundles Resolver](https://tekton.dev/docs/pipelines/bundle-resolver/)
1. Via the [`tektoncd/resolution` project](https://github.com/tektoncd/resolution)
1. Via the `taskRun.spec.taskRef.bundle` and the `pipelineRun.spec.pipelineRef.bundle` attributes
```

We can just use `1.` and let the Markdown renderer order it for us. That way we can't get the numbering wrong. 😉

And we can use a better link name.

> Renovate's Tekton manager supports all of them.

Reviewer: ```suggestion
Renovate's Tekton manager supports all the methods listed above.
```
